multivariate_normal_model.cpp

```diff
@@ -1,106 +1,52 @@
 #include "multivariate_normal_model.h"
 #include "opencv2/imgproc.hpp"
 
-MultivariateNormalModel::MultivariateNormalModel()
-    : is_trained_{false}
-{}
 
-bool MultivariateNormalModel::isTrained() const
-{ return is_trained_; }
+MultivariateNormalModel::MultivariateNormalModel(const cv::Mat& samples)
+{
+  performTraining(samples);
+}
+
 
 void MultivariateNormalModel::performTraining(const cv::Mat& samples)
 {
-  cv::Mat covar;
-  cv::calcCovarMatrix(samples, covar, mu_, cv::COVAR_NORMAL | cv::COVAR_COLS);
-  cv::invert(covar, inv_covar_, cv::DECOMP_SVD);
-
-  samples_ = samples.clone();
-  is_trained_ = true;
+  cv::calcCovarMatrix(samples, covariance_, mean_, cv::COVAR_NORMAL | cv::COVAR_COLS);
+  cv::invert(covariance_, inverse_covariance_, cv::DECOMP_SVD);
 }
 
+
 cv::Mat MultivariateNormalModel::computeMahalanobisDistances(const cv::Mat& image) const
 {
-  cv::Mat image_64;
-  image.convertTo(image_64, CV_64FC3);
+  constexpr double dist_to_uint8_scale = 1000.0;
 
-  cv::Mat mahalanobis_dist(image.size(), CV_64FC1);
+  cv::Mat mahalanobis_img(image.size(), CV_8UC1);
 
-  image_64.forEach<cv::Vec3d>(
-      [&](cv::Vec3d& pixel, const int pos[])
+  using Pixel = cv::Vec<uint8_t, 3>;
+  image.forEach<Pixel>(
+      [&](const Pixel& pixel, const int* pos)
       {
-        mahalanobis_dist.at<double>(pos[0], pos[1]) = cv::Mahalanobis(pixel, mu_, inv_covar_);
+        const cv::Vec3d double_pixel(pixel(0), pixel(1), pixel(2));
+        const double mahalanobis = cv::Mahalanobis(double_pixel, mean_, inverse_covariance_);
+        mahalanobis_img.at<uint8_t>(pos[0], pos[1]) = static_cast<uint8_t>(dist_to_uint8_scale * mahalanobis);
       });
 
-  return mahalanobis_dist;
+  return mahalanobis_img;
```

Comment:
Hva med å regne ut på samme måte som før, og i stedet for å skalere og caste inni løkka heller bruke `convertTo` etterpå?
```cpp
cv::Mat mahalanobis_img(image.size(), CV_64F);
image.forEach ...
{
 mahalanobis_img.at... = cv::Mahalanobis(double_pixel, mean_, inverse_covariance_);
}
cv::Mat scaled_img;
mahalanobis_img.convertTo(scaled_img, CV_8UC1, dist_to_uint8_scale);
```

For øvrig hadde jeg kanskje mer sansen for `image.convertTo(image_64, CV_64FC3);` enn `double_pixel`. Hva var grunnen til at du endret det?